### PR TITLE
3.3.21 Backport from Matrix

### PR DESCRIPTION
**Pull request by emveepee**
OpenFIle() was being called with wrong value for READ_NO_CACHE

### CODE REVIEW

File: src/buffers/ClientTimeshift.cpp [hunk 1]
```diff
@@ -84,7 +84,7 @@ bool ClientTimeShift::Open(const std::string inputUrl)
     return false;
   }
 
-  if (Buffer::Open(inputUrl, 0 ) == false)
+  if (Buffer::Open(inputUrl) == false)
```
Comment (phunkyfish) [inline src/buffers/ClientTimeshift.cpp]: `if (!Buffer::Open(inputUrl))`?

File: src/buffers/ClientTimeshift.cpp [hunk 2]
```diff
@@ -164,7 +164,7 @@ int64_t ClientTimeShift::Seek(int64_t position, int whence)
     m_streamPosition = position;
   }
   const std::string seekingInput = m_sourceURL + std::to_string(position ) + "-";
-  if ( Buffer::Open(seekingInput.c_str(), 0) == false)
+  if (Buffer::Open(seekingInput.c_str()) == false)
```
Comment (phunkyfish) [inline src/buffers/ClientTimeshift.cpp]: `if (!Buffer::Open(seekingInput.c_str()))`?

**Review by phunkyfish — APPROVED** [review]
Just some minors, up to you.

**Review by phunkyfish — APPROVED** [review]

**Comment by phunkyfish** [comment]
You really should squash these commits ;)

**Comment by emveepee** [comment]
> You really should squash these commits ;)

I wanted to highlite the additional change for the spaces around the / that you didn't request on this PR which came from your comments on PR 130.

**Comment by phunkyfish** [comment]
I must have missed that. Still fine to squash though 😉

**Review by phunkyfish — APPROVED** [review]
👍

**Comment by emveepee** [comment]
The backport to use no cache was not good enough, Kodi still did not handle no data on in-progress recordings well, especially on long seeks.   User testing indicates this commit works as expected.  Needs to be ported to Matrix.

**Review by phunkyfish — APPROVED** [review]

**Pull request merged**